### ISSUE TITLE
How can I run inference?

### ISSUE THREAD

**Issue by arnaud-secondlayer** [ximinng/LLM4SVG#18]
I have found pre-trained weights at https://huggingface.co/xingxm/LLM4SVG-GPT2XL-3B-Instruct-2401028
However, I'm not sure how to run inference with them.
Anybody managed to run inference with the pre-trained weights?

**Comment by ericaweng** [comment]
I loaded the model that you linked and inspected it; the tokenizer does not have the additional vocabulary added to it, and the generations look pretty bad. So, I'm guessing it's not a final version of the weights, only a new initialization. here are some examples:
```
test_prompts = [
        "a simple red circle",
        "a purple flower",
        "a minimalist house with door and windows",
    ]
```

<img width="150" alt="Image" src="https://github.com/user-attachments/assets/b6d91d5d-b839-4ed8-a718-e3b7101778ae" />

<img width="127" alt="Image" src="https://github.com/user-attachments/assets/65efe55c-cdcb-49ad-90d8-6c5193f71fd1" />

<img width="140" alt="Image" src="https://github.com/user-attachments/assets/9933b162-415d-457c-918d-b688c2274546" />

some generations are not even syntactically correct, which makes sense since the svg tokens were not added to the vocabulary.

**Comment by arnaud-secondlayer** [comment]
@ericaweng can you share the code for loading the weights and running inference?

**Comment by ericaweng** [comment]
Sorry, it was temporary code i generated with cursor that I already deleted. If i regenerate it I will update this post.

however, one thing you can do to see that the provided model is not trained is just inspect the json files at https://huggingface.co/xingxm/LLM4SVG-GPT2XL-3B-Instruct-2401028/tree/main. e.g., if you open up tokenizer_config.json, you'll notice it doesn't contain any svg tokens.

**Comment by arnaud-secondlayer** [comment]
Can someone of the project help with this? Are we expected to get results with quality on par with the samples in the paper with these weights?
As far as I can see in the project issues, it looks like nobody has been able to reproduce the paper results.

**Comment by olgacanva** [comment]
I run training commands with Qwen and GPT models but also experienced loss spike even on recommended 8 A100 GPUs. I'm keen to know as well if someone is able to reproduce results.

**Comment by Eric-zhang-1111** [comment]
> Sorry, it was temporary code i generated with cursor that I already deleted. If i regenerate it I will update this post.
> 
> however, one thing you can do to see that the provided model is not trained is just inspect the json files at https://huggingface.co/xingxm/LLM4SVG-GPT2XL-3B-Instruct-2401028/tree/main. e.g., if you open up tokenizer_config.json, you'll notice it doesn't contain any svg tokens.

Hi, I found this model contains svg tokens
https://huggingface.co/xingxm/LLM4SVG-DeepSeekDistill-Qwen-Instruct-2401028/blob/main/tokenizer_config.json

**Comment by loevlie** [comment]
So no one can replicate the results?
